airflow-core/src/airflow/models/dag.py

```diff
@@ -385,6 +385,7 @@ class DAG(TaskSDKDag, LoggingMixin):
     :param dagrun_timeout: Specify the duration a DagRun should be allowed to run before it times out or
         fails. Task instances that are running when a DagRun is timed out will be marked as skipped.
     :param sla_miss_callback: DEPRECATED - The SLA feature is removed in Airflow 3.0, to be replaced with a new implementation in 3.1
+    :param deadline: Optional Deadline Alert for the DAG.
```

Comment:
nit:
more context for the new exciting feature (i find myself often looking up and refering the docstring-generated airflow documentation online)
```suggestion
    :param deadline: Optional Deadline Alert for the DAG. Specifies a time by which
    the DAG run should complete, calculated relative to a reference timestamp.
    If the deadline passes before completion, the configured callback is triggered.
    Example: DeadlineAlert(reference=DeadlineReference.DAGRUN_LOGICAL_DATE,
    interval=timedelta(hours=1), callback=my_callback)
```

Reply:
Interesting.  I'll add it.  I'm in the habit of making the docstrings as succinct as possible,. but that doesn't mean it's a good habit.   I'll make the change.  Thanks.

Reply:
Added a slight variation on this in [11c7872](https://github.com/apache/airflow/pull/50093/commits/11c7872818e13731cb5f3f046b47cd1893eece36).